cmd/kubelet/app/server.go

```diff
@@ -306,8 +304,18 @@ func initConfigz(kc *componentconfig.KubeletConfiguration) (*configz.Config, err
 }
 
 func run(s *options.KubeletServer, kubeDeps *kubelet.KubeletDeps) (err error) {
-	// TODO: this should be replaced by a --standalone flag
-	standaloneMode := (len(s.APIServerList) == 0 && !s.RequireKubeConfig)
+
+	// Deprecation warning and guidance for --require-kubeconfig
+	if s.RequireKubeConfig.Provided() {
+		if s.RequireKubeConfig.Value() == true {
+			glog.Warningf("--require-kubeconfig is deprecated. Please enable API server mode by setting --kubeconfig instead of using --require-kubeconfig=true.")
+		} else {
+			glog.Warningf("--require-kubeconfig is deprecated. Please enable standalone mode by simply omitting both --kubeconfig and --require-kubeconfig.")
+		}
+	}
+
+	standaloneMode := (s.RequireKubeConfig.Provided() && s.RequireKubeConfig.Value() == false) || // --require-kubeconfig=false
```

Comment:
might be simpler to start this in `true`, then flip it to false in specific cases:

```
standaloneMode := true
switch {
case s.RequireKubeConfig.Provided() && s.RequireKubeConfig.Value() == true:
  standaloneMode = false
  glog.Warningf("--require-kubeconfig is deprecated. Set --kubeconfig instead of using --require-kubeconfig=true.")

case s.RequireKubeConfig.Provided() && s.RequireKubeConfig.Value() == false:
  glog.Warningf("--require-kubeconfig is deprecated. Omit --kubeconfig instead of using --require-kubeconfig=false.")

case s.KubeConfig.Provided():
  standaloneMode = false
}
```